include/experimental/__p2630_bits/submdspan_extents.hpp

```diff
@@ -52,6 +53,33 @@ template <class OffsetType, class ExtentType, class StrideType>
 struct is_strided_slice<
     strided_slice<OffsetType, ExtentType, StrideType>> : std::true_type {};
 
+// Helper for identifying valid pair like things
+template <class T, class IndexType> struct index_pair_like : std::false_type {};
+
+template <class IdxT1, class IdxT2, class IndexType>
+struct index_pair_like<std::pair<IdxT1, IdxT2>, IndexType> {
+  static constexpr bool value = std::is_convertible_v<IdxT1, IndexType> &&
+                                std::is_convertible_v<IdxT2, IndexType>;
+};
+
+template <class IdxT1, class IdxT2, class IndexType>
+struct index_pair_like<std::tuple<IdxT1, IdxT2>, IndexType> {
+  static constexpr bool value = std::is_convertible_v<IdxT1, IndexType> &&
+                                std::is_convertible_v<IdxT2, IndexType>;
+};
+
+template <class IdxT, class IndexType>
+struct index_pair_like<std::complex<IdxT>, IndexType> {
+  static constexpr bool value = std::is_convertible_v<IdxT, IndexType>;
+};
```

Comment:
`std::get<0>(z)` is only available from C++26 on, right? https://en.cppreference.com/w/cpp/numeric/complex/get

Reply:
It also wouldn't work portably. In fact we need to overhaul more stuff because `integral_constant::value()` and `get<>(tuple)` give warnings.

Reply:
And just to be clear: the mdspan specification uses `pair-like`,  and that is a named list of types, not everything which supports tuple protocol and has size 2. Hence, my change in the implementation here.

Reply:
It is not because it will happen to work in C++26 for complex that we should get out of our way, pull in the header for `<complex>` and specialize.
This is not necessary for the Kokkos refactor.  Please withdraw the header include and that specialization and propose it as a follow up if you really want it.

Reply:
At least `std::get` isn't used with `complex` anymore after the last commits.